maintenance/health/servicehealthcheck/background_healthchecker.go

```diff
@@ -0,0 +1,104 @@
+package servicehealthcheck
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+
+	"github.com/pace/bricks/maintenance/errors"
+	"github.com/pace/bricks/maintenance/log"
+)
+
+// BackgroundHealthCheck is an interface that, when implemented, will run the
+// HealthCheck in the background, and cache the result, that will be fetched by the HealthHandler
+type BackgroundHealthCheck interface {
```

Comment:
I assume you did this in the assumption that we want to give each health check the option of defining its own intervals. However, that would require us to change every health check implementation to support the Interval() function.

Instead, I propose to add Options to the `RegisterHealthCheck` and `RegisterOptionalHealthCheck` functions that give the owner the option of setting an interval, and using a struct to store it here.

Reply:
Added the Options. 

Although, one thing I'm not sure of, is the use of the MaxWait in the HealthChecks called from the Handler. 
Basically the `cfg.HealthCheckMaxWait` is the default, and each HealthCheck can have its own, which is fine for background ones, but might not be fine for those called from the Handler, as the bigger maxWait becomes a maxWait for the entire Handler.

Although, if not using it this way, and having the `cfg.HealthCheckMaxWait` as a strong timeout on the Handler, then `UseMaxWait` is an option usable for background HealthChecks only, which could look a bit strange from the package API point of view.